nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,109 @@
-import org.nuxeo.ecm.core.api.DocumentModel;
-import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriterTest;
-import org.nuxeo.ecm.core.io.marshallers.json.JsonAssert;
-import org.nuxeo.ecm.core.io.marshallers.json.document.DocumentModelJsonWriter;
-import org.nuxeo.ecm.core.io.registry.context.RenderingContext;
-import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.io.registry.MarshallingConstants;
 import org.nuxeo.ecm.core.test.annotations.Granularity;
 import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.restapi.test.RestServerFeature;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.jaxrs.test.HttpClientTestRule;
 import org.nuxeo.runtime.test.runner.Deploy;
 import org.nuxeo.runtime.test.runner.Features;
 import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.ServletContainerFeature;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.jersey.core.util.MultivaluedMapImpl;
 
 @RunWith(FeaturesRunner.class)
-@Features(CoreFeature.class)
+@Features(RestServerFeature.class)
 @RepositoryConfig(cleanup = Granularity.METHOD)
 @Deploy("org.nuxeo.ecm.platform.video:OSGI-INF/core-types-contrib.xml")
 @Deploy("org.nuxeo.ecm.platform.audio.core:OSGI-INF/core-types-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.tag:OSGI-INF/faceted-tag-service-core-types.xml")
 @Deploy("org.nuxeo.ecm.platform.picture.core:OSGI-INF/picture-schemas-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.types:OSGI-INF/subtypes-enricher-contrib.xml")
 @Deploy("org.nuxeo.compound.documents")
-public class TestCompoundDocument extends AbstractJsonWriterTest.Local<DocumentModelJsonWriter, DocumentModel> {
+public class TestCompoundDocument {
 
     @Inject
-    protected CoreSession session;
-    
-    public TestCompoundDocument() {
-        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected HttpClientTestRule httpClientRule;
+
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    protected Set<String> subtypes = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+
+    @Before
+    public void before() {
+        httpClientRule = new HttpClientTestRule.Builder().credentials("Administrator", "Administrator")
+                                                         .url(String.format("http://localhost:%d/api/v1",
+                                                                 servletContainerFeature.getPort()))
+                                                         .build();
+        httpClientRule.starting();
     }
 
+    @After
+    public void after() {
+        httpClientRule.finished();
+    }
+
+    // Avoid useless warning for not finding dclistener
+    @Deploy("org.nuxeo.compound.documents:OSGI-INF/disable-dclistener-contrib.xml")
     @Test
```

Comment:
```suggestion
    @Test
    @Deploy("org.nuxeo.compound.documents:OSGI-INF/disable-dclistener-contrib.xml")
```

Reply:
In fact, just after checking the listener, you shouldn't disable it. We're trying to test the real case where this listener is enabled.